src/service/url-replacements-impl.js

```diff
@@ -445,94 +459,166 @@ export class UrlReplacements {
    * The data for the timing events is retrieved from performance.timing API.
    * If start and end events are both given, the result is the difference between the two.
    * If only start event is given, the result is the timing value at start event.
-   * @param {*} startEvent
-   * @param {*=} endEvent
-   * @return {!Promise<string|undefined>}
+   * @param {string} startEvent
+   * @param {string=} endEvent
+   * @return {!Promise<ResolverReturnDef>}
    * @private
    */
-  getTimingData_(startEvent, endEvent) {
-    const timingInfo = this.win_['performance']
-        && this.win_['performance']['timing'];
+  getTimingDataAsync_(startEvent, endEvent) {
+    let metric = this.getTimingDataSync_(startEvent, endEvent);
```

Comment:
I call sync first so that timing API information that does not require waiting for load (e.g. navigation, DOM content loaded, etc) do not need to unnecessarily wait for load.  Ideally this function would wait for to the correct event depending upon the information requested (e.g. DOMContentLoaded vs. Load).